CRM/Civicase/Service/CaseRoleCreationPreProcess.php

```diff
@@ -16,7 +16,11 @@ class CRM_Civicase_Service_CaseRoleCreationPreProcess extends CRM_Civicase_Servi
    *   API request parameters.
    */
   public function onCreate(array &$requestParams) {
-    if ($this->isSingleCaseRole && !$this->isMultiClient) {
+    if (
+      ($this->isSingleCaseRole && !$this->isMultiClient) &&
+      (!isset($requestParams['params']['start_date']) ||
+      empty($requestParams['params']['start_date']))
```

Comment:
I think it will be fine if we just use `empty($requestParams['params']['start_date'])` here. Empty is equivalent to ` !isset($var) || $var == false.`

https://www.php.net/manual/en/function.empty.php